File: Pascal'sTriangle.py
class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        
        res = []
        for row in range(numRows):
            #The digits of 11**0, ..., 11**4 are the same as the values in rows 0 through 4
            if row <= 4:
                res.append([int(i) for i in str(11**row)])
                continue
            
            #Afterwards, initialize the row as all ones since the first and last value will always be 1
            temp = [1]*(row+1)
            
            #Then fill in indices 1 through -2 with the sum of the two values directly above it
            for col in list(range(1,row)):
                temp[col] = res[row-1][col-1] + res[row-1][col]
                
            res.append(temp)
                    
        return res

File: Pascal'sTriangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int numRows) 
    {
        vector<vector<int>> result;
        
        
        for(int i = 0; i < numRows; i++)
        {
            if(i == 0)
            {
                vector<int> temp;
                temp.push_back(1);
                result.emplace_back(temp);
            }
            
            if(i == 1)
            {
                vector<int> temp;
                temp.push_back(1);
                temp.push_back(1);
                result.emplace_back(temp);
            }
            
            if(i > 1)
            {
                vector<int> temp;
                
                temp.push_back(1);
                
                int size = result[result.size() - 1].size();
                
                for(int i = 1; i < size; i++)
                {
                    int sum = result[result.size() - 1][i] + result[result.size() - 1][i - 1];
                    temp.push_back(sum);
                }
                
                temp.push_back(1);
                
                result.emplace_back(temp);
                
            }
            
        }
        
        return result;
        
    }
};